metricbeat/module/redis/metricset.go

```diff
@@ -39,21 +42,38 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 		IdleTimeout time.Duration `config:"idle_timeout"`
 		Network     string        `config:"network"`
 		MaxConn     int           `config:"maxconn" validate:"min=1"`
-		Password    string        `config:"password"`
 	}{
-		Network:  "tcp",
-		MaxConn:  10,
-		Password: "",
+		Network: "tcp",
+		MaxConn: 10,
 	}
 	err := base.Module().UnpackConfig(&config)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to read configuration")
 	}
 
+	uriParsed, _ := url.Parse(base.HostData().URI)
+	if uriParsed.RawQuery != "" {
+		queries := strings.Split(uriParsed.RawQuery, "&")
+		pw := ""
+		db := 0
+		for _, query := range queries {
+			querySplit := strings.Split(query, "=")
+			if querySplit[0] == "password" {
```

Comment:
Maybe I missed it, but I didn't see a test case with `password` in the query string. I think it would be good to test with a query string that both has `password` and does not have `password`.

Reply:
yep, tests will be added in next commit 👍